front/components/assistant_builder/ActionsScreen.tsx

```diff
@@ -509,6 +509,116 @@ function ActionCard({
   );
 }
 
+function ActionConfigEditor({
+  owner,
+  action,
+  dustApps,
+  dataSources,
+  instructions,
+  updateAction,
+  setEdited,
+}: {
+  owner: WorkspaceType;
+  action: AssistantBuilderActionConfiguration;
+  dustApps: AppType[];
+  dataSources: DataSourceType[];
+  instructions: string | null;
+  updateAction: (args: {
+    actionName: string;
+    actionDescription: string;
+    getNewActionConfig: (
+      old: AssistantBuilderActionConfiguration["configuration"]
+    ) => AssistantBuilderActionConfiguration["configuration"];
+  }) => void;
+  setEdited: (edited: boolean) => void;
+}) {
```

Comment:
`nit`

Extract this to a dedicated interface.

Reply:
Might do, but not here